source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -50,11 +54,133 @@ void Filter::onRead() {
 
 void Filter::onReadWorker() {
   Network::ConnectionSocket& socket = cb_->socket();
-  std::string proxy_line;
-  if (!readLine(socket.fd(), proxy_line)) {
+
+  if ((!proxyProtocolHeader_.has_value() && !readProxyHeader(socket.fd())) ||
+      (proxyProtocolHeader_.has_value() && proxyProtocolHeader_.value().extensions_length_ &&
+       !parseExtensions(socket.fd()))) {
```

Comment:
Seems like we could leave out the `proxyProtocolHeader_.value().extensions_length_` check here since `parseExtensions()` is a while loop over `proxyProtocolHeader_.value().extensions_length_`.